### PR TITLE
Fix grad_norm unserializable tensor log failure

### PR DESCRIPTION
**Pull request by svenschultze**
# What does this PR do?
Fixes Issue #29207 

@muellerzr 
@pacman100

### CODE REVIEW

**Comment by svenschultze** [comment]
The issue seems to be that deepspeeds get_global_grad_norm() unexprectedly returns a tensor instead of a float.
This fixes that issue. For Non-Deepspeed training, we already call item(), so there is no fix required.

**Review by qianhang1994 — APPROVED** [review]
met same error too, here is the variable print log
[json_string = json.dumps(dataclasses.asdict(self), indent=2, sort_keys=True) + "\n"](https://github.com/huggingface/transformers/blob/main/src/transformers/trainer_callback.py#L113)
"print(dataclasses.asdict(self))" results:

> 
>  {...
> 'log_history': [
> {'loss': 24.6648, 'grad_norm': tensor(1091.0826), 'learning_rate': 1e-05, 'epoch': 0.04, 'step': 5}, 
> {'loss': 8.7945, 'grad_norm': tensor(248.9097), 'learning_rate': 1e-05, 'epoch': 0.08, 'step': 10}
> ], ...}

show grad_norm is a tensor

**Review by muellerzr — APPROVED** [review]
Thanks, looks good to me!

**Comment by muellerzr** [comment]
Actually, this seems to be in direct conflict with https://github.com/huggingface/transformers/pull/29158

**Comment by amyeroberts** [comment]
@muellerzr In this case, which PR should stand?

**Comment by winglian** [comment]
> Actually, this seems to be in direct conflict with #29158

That other PR seems to affect the `else` clause where, whereas this is in the `if` part of the clause, so it might be okay

**Comment by muellerzr** [comment]
This one should stand and be okay, the other PR will need to change, so this is fine to me

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing!
